### ISSUE TITLE
Magic method is not working

### ISSUE THREAD

**Issue by chief-kh** [phalcon/cphalcon#10885]
Table structure:

```php
CREATE TABLE `plant` (
  `id` int(10) unsigned NOT NULL AUTO_INCREMENT,
  `title` varchar(255) NOT NULL,
  PRIMARY KEY (`id`),
  UNIQUE KEY `plant_title` (`title`)
) ENGINE=MyISAM AUTO_INCREMENT=118 DEFAULT CHARSET=utf8;
```

Model code:

```
class Plant extends \Phalcon\Mvc\Model {

    public $id;
    public $title;

    protected function beforeSave() {
        if(empty($this->id) && $plant = self::findFirstByTitle($this->title)) {
            $this->id = $plant->id;
            return false;
        }
    }
}
```

PhalconException: The method 'findFirstByTitle' doesn't exist on model 'Plant'

**Comment by andresgutierrez** [comment]
It looks like calling a static method invokes __call instead of __callStatic that is what makes the magic.

**Comment by andresgutierrez** [comment]
Fixed in the 2.0.x branch
